drivers/digital-io/max22200/max22200.c

```diff
@@ -0,0 +1,958 @@
+		  struct max22200_init_param *init_param)
+{
+	struct max22200_desc *descriptor;
+	uint32_t reg_val;
+	uint32_t status_reg = 0;
+	int ret, i;
+
+	descriptor = no_os_calloc(1, sizeof(*descriptor));
+	if (!descriptor)
+		return -ENOMEM;
+
+	ret = no_os_spi_init(&descriptor->comm_desc, init_param->comm_param);
+	if (ret)
+		goto err;
+
+	/* Time between enable pin set and device power-up. */
+	no_os_udelay(500);
+
+	ret = no_os_gpio_get(&descriptor->cmd_desc,
+			     init_param->cmd_param);
+	if (ret)
+		goto spi_err;
+
+	ret = no_os_gpio_direction_output(descriptor->cmd_desc, NO_OS_GPIO_LOW);
+	if (ret)
+		goto cmd_err;
+
+	/* Enable pin is optional to be connected to master through GPIO and be
+	   controlled in this way, but mandatory to be logic high (3.3V) in
+	   order for the device to be enabled. */
+	ret = no_os_gpio_get_optional(&descriptor->enable_desc,
+				      init_param->enable_param);
+	if (ret)
+		goto cmd_err;
+
+	if (descriptor->enable_desc) {
+		ret = no_os_gpio_direction_output(descriptor->enable_desc,
+						  NO_OS_GPIO_HIGH);
+		if (ret)
+			goto enable_err;
+	}
+
+	ret = no_os_gpio_get_optional(&descriptor->trig_desc,
+				      init_param->trig_param);
+	if (ret)
+		goto enable_err;
+
+	/* External trigger if used, is set logic LOW at initialization.
+	   Can be set logic HIGH in case of choosing to use the external
+	   trigger for Full-Bridge Mode. */
+	if (descriptor->trig_desc) {
+		ret = no_os_gpio_direction_output(descriptor->trig_desc,
+						  NO_OS_GPIO_LOW);
+		if (ret)
+			goto trig_err;
+	}
+
+	ret = no_os_gpio_get_optional(&descriptor->fault_desc,
+				      init_param->fault_param);
+	if (ret)
+		goto trig_err;
+
+	if (descriptor->fault_desc) {
+		ret = no_os_gpio_direction_input(descriptor->fault_desc);
+		if (ret)
+			goto fault_err;
+	}
+
+	ret = max22200_reg_read(descriptor, MAX22200_STATUS_REG, &reg_val,
+				false);
+	if (ret)
+		goto fault_err;
+
+	/* Getting channel mode info from init_param */
+	for (i = 0; i < MAX22200_CHANNELS_CONFIG; i++)
+		status_reg |= no_os_field_prep(MAX22200_CH_MODE_MASK(i),
+					       *(init_param->ch_config + i));
+
+
+	/* Writing to the status register the mode of the channels and
+	   setting active bit to 1. */
+	status_reg |= no_os_field_prep(MAX22200_ACTIVE_MASK, 1);
+	ret = max22200_reg_write(descriptor, MAX22200_STATUS_REG,
+				 status_reg, false);
+	if (ret)
+		goto fault_err;
+
+	/* Configuring channels at startup. */
+	for (i = 0; i < MAX22200_CHANNELS; i++) {
+		ret = max22200_set_ch_scale(descriptor, i, MAX22200_FULLSCALE);
+		if (ret)
+			goto fault_err;
+
+		ret = max22200_set_ch_hold(descriptor, i, MAX22200_HIT_MAX_VAL);
+		if (ret)
+			goto fault_err;
+
+		ret = max22200_set_ch_trig(descriptor, i, MAX22200_ONCH_SPI);
+		if (ret)
+			goto fault_err;
+
+		ret = max22200_set_ch_hit(descriptor, i, MAX22200_HIT_MAX_VAL,
+					  MAX22200_HIT_NO_TIME);
+		if (ret)
+			goto fault_err;
+	}
+
+	no_os_udelay(2500);
```

Comment:
meaning of this delay?

Reply:
Adding a comment specifying why this delay is needed.